nnfabrik/templates/trained_model.py

```diff
@@ -2,29 +2,104 @@
 import tempfile
 import torch
 import os
-from nnfabrik.main import Model, Dataset, Trainer, Seed, Fabrikant
-from nnfabrik.builder import get_all_parts, get_model, get_trainer
-from nnfabrik.utility.dj_helpers import gitlog, make_hash
-from .utility import DataInfoBase
+from ..builder import get_all_parts, get_model, get_trainer
+from . import main
+from ..utility.dj_helpers import make_hash
 from datajoint.fetch import DataJointError
 import warnings
+from types import ModuleType
+from typing import Optional, Union
 
 
 class TrainedModelBase(dj.Computed):
     """
-    Inherit from this class and decorate with your own schema to create a functional
-    TrainedModel table. By default, this will inherit from following
-    Model, Dataset, Trainer and Seed as found in nnfabrik.main, and also point to . To change this behavior,
-    overwrite the `model_table`, `dataset_table`, `trainer_table` and `seed_table` class
-    properties.
+    Base class for defining TrainedModel table used to tigger training of models in nnfabrik.
+
+    To use this class, define a new class inheriting from this base class, and decorate with your own
+    schema. Furthermore, you have to do one of the following for the class to be functional:
+    * Set the class property `nnfabrik` to point to a module or a dictionary context that contains classes
+        for tables corresponding to `Fabrikant`, `Seed`, `Dataset`, `Model`, and `Trainer`. Most commonly, you
+        would want to simply pass the resulting module object from `my_nnfabrikant` output.
```

Comment:
I believe it's `my_nnfabrik`, and not `my_nnfabrikant` 😉

Reply:
Good catch